common/k8s/src/middleware/runner.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{
+    create_k8s_client_default_from_env,
+    middleware::{K8sMetadata, StoreSwapIntent},
+};
+use futures::StreamExt;
+use hyper::http::header::HeaderValue;
+use middleware::Executor;
+use std::sync::Arc;
+use tokio::{
+    task::JoinHandle,
+    time::{sleep, Duration},
+};
+use tracing::{error, trace, warn};
+
+pub static K8S_WATCHER_ROTATION_DELAY: Duration = Duration::from_secs(3);
+
+enum State<'a> {
+    Init,
+    Creating(bool),
+    Kicking(bool, kube::Client),
+    Swapping(bool, JoinHandle<()>, StoreSwapIntent<'a>),
+    Running(JoinHandle<()>),
+}
+
+async fn step_trampoline<'a>(
```

Comment:
I added exponential backoff, it will pause after the Running state. It starts at 2 seconds and goes to a maximum of 30 seconds. If the running state is able to run successfully for longer than 30 seconds, the attempts counter will get reset and go back to a 2 second delay.